prog/kubernetes/provision_kubernetes_node.rb

```diff
@@ -7,6 +7,10 @@ def vm
     @vm ||= Vm[frame.fetch("vm_id", nil)] || nil
   end
 
+  def kubernetes_nodepool
+    @kubernetes_nodepool ||= KubernetesNodepool[frame.fetch("nodepool_id", nil)] || nil
```

Comment:
Since you give `nil` as default value

```suggestion
    @kubernetes_nodepool ||= KubernetesNodepool[frame["nodepool_id"]] || nil
```